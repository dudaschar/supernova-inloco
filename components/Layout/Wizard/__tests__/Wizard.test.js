import { shallow } from 'enzyme'
import React from 'react'

import LayoutWizard from '../'

const buildWrapper = props =>
  shallow(
    <LayoutWizard
      className="customTestClass"
      steps={['Step 1', 'Step 2', 'Step 3']}
      {...props}>
      Test Child Content
    </LayoutWizard>
  )

describe('LayoutWizard', () => {
  it('should render layout wizard content for the first step by default', () => {
    expect(buildWrapper()).toMatchSnapshot()
  })

  it('should render layout wizard content for the given initial step', () => {
    const wrapper = buildWrapper({ initialStepIndex: 1 })
    expect(wrapper).toMatchSnapshot()
  })
})
