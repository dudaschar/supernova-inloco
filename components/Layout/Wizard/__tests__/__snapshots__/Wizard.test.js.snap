// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutWizard should render layout wizard content for the first step by default 1`] = `
<Fragment>
  <Header>
    <HeaderTitle>
      <div
        className="inloco-layout__wizard-title"
      >
        <div
          className="inloco-layout__wizard-title-number"
        >
          1
        </div>
        Step 1
      </div>
    </HeaderTitle>
  </Header>
  <MainContent>
    <Wizard
      className="inloco-layout__wizard customTestClass"
      currentStep={0}
      footerProps={Object {}}
      onChangeStep={[Function]}
      steps={
        Array [
          "Step 1",
          "Step 2",
          "Step 3",
        ]
      }
    >
      Test Child Content
    </Wizard>
  </MainContent>
</Fragment>
`;

exports[`LayoutWizard should render layout wizard content for the given initial step 1`] = `
<Fragment>
  <Header>
    <HeaderTitle>
      <div
        className="inloco-layout__wizard-title"
      >
        <div
          className="inloco-layout__wizard-title-number"
        >
          2
        </div>
        Step 2
      </div>
    </HeaderTitle>
  </Header>
  <MainContent>
    <Wizard
      className="inloco-layout__wizard customTestClass"
      currentStep={1}
      footerProps={Object {}}
      onChangeStep={[Function]}
      steps={
        Array [
          "Step 1",
          "Step 2",
          "Step 3",
        ]
      }
    >
      Test Child Content
    </Wizard>
  </MainContent>
</Fragment>
`;
